extensions/metadata.md

```diff
@@ -0,0 +1,740 @@
+| 761 | `RPL_KEYVALUE`            | `<Target> <Key> <Visibility>[ :<Value>]` |
+| 766 | `RPL_KEYNOTSET`           | `<Target> <Key> :key not set`            |
+| 770 | `RPL_METADATASUBOK`       | `:<Key1> [<Key2> ...]`                   |
+| 771 | `RPL_METADATAUNSUBOK`     | `:<Key1> [<Key2> ...]`                   |
+| 772 | `RPL_METADATASUBS`        | `:<Key1> [<Key2> ...]`                   |
+| 774 | `RPL_METADATASYNCLATER`   | `<Target> [<RetryAfter>]`                |
+
+Reference table of numerics and the `METADATA` subcommands or any other commands that produce them:
+
+| Label                              | GET | LIST | SET | CLEAR | SUB | UNSUB | SUBS | SYNC | Other   |
+| ---    | :-: | :-: | :-: | :-: | :-: | :-: | :-: | :-: | :-: |
+| `RPL_WHOISKEYVALUE`                |     |      |     |       |     |       |      |      | `WHOIS` |
+| `RPL_KEYVALUE`                     | *   | *    | *   | *     |     |       |      |      |         |
+| `RPL_KEYNOTSET`                    | *   |      |     |       |     |       |      |      |         |
+| `RPL_METADATASUBOK`                |     |      |     |       | *   |       |      |      |         |
+| `RPL_METADATAUNSUBOK`              |     |      |     |       |     | *     |      |      |         |
+| `RPL_METADATASUBS`                 |     |      |     |       |     |       | *    |      |         |
+| `RPL_METADATASYNCLATER`            |     |      |     |       | *   |       |      | *    | `JOIN`  |
+| `RPL_METADATASUBS`                 |     |      |     |       |     |       | *    |      |         |
+
+Replies:
+
+* `RPL_KEYVALUE` reports the values of metadata keys. The `Visibility` parameter is defined in the [server message](#metadata-server-message) section.
+
+### `RPL_WHOISKEYVALUE` numeric
+
+When a user runs `WHOIS` on a user with metadata, a subset of that metadata MAY be sent with `RPL_WHOISKEYVALUE` numerics. This subset MUST be chosen explicitly, and optimised for keys that can be easily read by users. For a complete view of user metadata, see the [`LIST`](#metadata-list) subcommand.
+
+## Examples
+
+These examples show the labels of the numerics (e.g. `RPL_METADATASUBOK`) instead of their number (e.g. `775`) in order to aid understanding. In a real implementation, messages always contain the number of the numeric.
+
+All examples begin with the client not being subscribed to any keys.
+
+-----
+
+### Capability Examples
+
+#### Capability value in `CAP LS` 1 
+
+    C: CAP LS 302
+    S: CAP * LS :userhost-in-names draft/metadata-2=foo,maxsub=50,bar multi-prefix
+
+#### Capability value in `CAP LS` 2
+
+    C: CAP LS 302
+    S: CAP * LS :draft/metadata-2=maxsub=25 multi-prefix invite-notify
+
+-----
+
+### METADATA command examples
+
+#### Setting metadata on self
+
+    C: METADATA * SET url :http://www.example.com
+    S: :irc.example.com RPL_KEYVALUE * url * :http://www.example.com
```

Comment:
The numeric names in these examples should be replaced with the actual numeric values